### PR TITLE
[doc] Add a cryptolib API documentation page.

### PR DESCRIPTION
**Pull request by jadephilipoom**
Translate the cryptolib API document to website format, including infrastructure to pull code snippets directly from header files. The prose here is taken directly from the preexisting document and I tried to convert the format as faithfully as possible.

The header-snippet infrastructure will automatically keep the code and the documentation in sync. It lets you write something with the file name and the name of the declaration (struct, function, enum, or typedef) that you want to pull:
```md
{{#header-snippet sw/device/lib/crypto/include/hash.h otcrypto_hash }}
```
When you compile the documentation, the `mdbook_code_snippet.py` preprocessor looks through the header file and pulls both the definition and any `/** ... */` comment that immediately precedes it. It adds a code block and also a link to the relevant line of source code on GitHub. The result looks like this:
Screenshot from 2023-07-27 08-26-33

This should work for non-cryptolib things (e.g. difs) as well, although I haven't applied it to any other docs pages. You can view it locally by:
- checking out this branch
- running `util/build-docs.sh serve`
- navigating to http://localhost:9000/book/doc/security/cryptolib/cryptolib_api.html

### CODE REVIEW

**Comment by moidx** [comment]
+ @hcallahan-lowrisc for visibility on mdbook related changes.

**Review by y-srini — APPROVED** [review]
Thanks for adding the cryptolib API doc. Looks good to me @jadephilipoom.

**Review by HU90m — APPROVED** [review]
Ran locally. Looks brilliant!

Only suggestion is to make any document being linked to from the documentation public.

File: doc/security/cryptolib/cryptolib_api.md [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+[nist-sp800-57]: https://csrc.nist.gov/publications/detail/sp/800-57-part-1/rev-5/final
+[otbn]: ../../../hw/ip/otbn/README.md
+[rsa-rfc]: https://datatracker.ietf.org/doc/html/rfc8017
+[safe-curves]: https://safecurves.cr.yp.to/
+[sec1]: https://www.secg.org/sec1-v2.pdf
+[sec2]: https://www.secg.org/sec2-v2.pdf
+[sha2-spec]: https://csrc.nist.gov/publications/detail/fips/180/4/final
+[sha3-spec]: https://csrc.nist.gov/publications/detail/fips/202/final
+[sha3-derived-spec]: https://csrc.nist.gov/publications/detail/sp/800-185/final
+[use-case-table]: https://docs.google.com/document/d/1fHUUL4i39FJXk-lbVNDizVZSWObizGE2pTWVhQKb41c
```
[the remark this answers is not in the document]
Comment (jadephilipoom) [reply doc/security/cryptolib/cryptolib_api.md]: Removed all references to non-public documents (this one and one other one I found); another option would be converting them to documentation pages in their own right, but I think by now both are somewhat outdated and probably not super useful to maintain. So I added a little extra exposition to this page where appropriate and removed the links.

[the remark this answers is not in the document]
Comment (HU90m) [reply doc/security/cryptolib/cryptolib_api.md]: Perfect, thank you!

**Review by moidx — APPROVED** [review]

**Pull request merged**